sw/otbn/crypto/p384_base.s

```diff
@@ -339,6 +367,78 @@ p384_mulmod_n:
   /* return result: c =[w17, w16] =  a * b % m. */
   ret
 
+/**
+ * 384-bit modular multiplication based on Solinas reduction algorithm.
+ *
+ * Returns c = a * b mod m.
+ *
+ * This subroutine is intended for use with the group order (n) of P-384, but
+ * will work for any modulus m such that 2^384 - 2^191 < m < 2^384.
+ *
+ * For mor information on the reduction algorith, see 'p384_reduce_n'.
+ *
+ * This routine runs in constant time.
+ *
+ * Flags: Flags have no meaning beyond the scope of this subroutine.
+ *
+ * @param[in] [w11, w10]: a, first operand, max. length 384 bit, b < m.
+ * @param[in] [w17, w16]: b, second operand, max. length 384 bit, b < m.
+ * @param[in] [w13, w12]: m, modulus, 2^383 <= m < 2^384.
+ * @param[in] w14: k, Solinas constant (2^384 - modulus), max. length 191 bit.
+ * @param[in] w31: all-zero.
+ * @param[out] [w17, w16]: c, result, max. length 384 bit.
+ *
+ * Clobbered registers: w16 to w24
+ * Clobbered flag groups: FG0
+ */
+.globl p384_mulmod_n
+p384_mulmod_n:
+  /* Compute the raw 768-bit product:
+       ab = [w20:w18] <= a * b */
+  jal     x1, mul384
+
+  /* [w17, w16] = ab mod m = [w20:w18] mod m */
+  jal     x1, p384_reduce_n
+
+  /* return result: c = [w17, w16] =  a * b mod m. */
+  ret
```

Comment:
nit: this could be a tail call to make it slightly faster (the `ret` from `p384_reduce_n` will now return to the caller of `p384_mulmod_n`). Same applies to the routine below.
```suggestion
  /* [w17, w16] = ab mod m = [w20:w18] mod m */
  jal     x0, p384_reduce_n
```

Reply:
Thanks for the hint! I'll change it.